src/customHeap.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * @license MIT
+ * @copyright 2020 Eyas Ranjous <eyas.ranjous@gmail.com>
+ */
+
+const { Heap } = require('./heap');
+
+/**
+ * @class CustomHeap
+ * @extends Heap
+ */
+class CustomHeap extends Heap {
```

Comment:
Not really sure if this new class is necessary. Why can't we just make the existing Heap accept a new compare function?

Mostly Javascript code is bundled up in the client application. And the size matters. IMO, this new class is redundant. And as a user for this library. I would strongly suggest having just a custom compare function instead of creating an entire new class.